daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -372,8 +373,17 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
     def finish: Result[(SubmittedTransaction, Tx.Metadata)] =
       machine.finish match {
         case Right(
-              UpdateMachine.Result(tx, _, nodeSeeds, globalKeyMapping, processedDisclosedContract)
+              UpdateMachine.Result(tx, _, nodeSeeds, globalKeyMapping, disclosedCreateEvents)
             ) =>
+          val disclosureMap = disclosures.iterator.map(c => c.contractId.value -> c).toMap
+          val processedDisclosedContracts = disclosedCreateEvents.map { create =>
+            val diclosedContract = disclosureMap(create.coid)
+            ProcessedDisclosedContract(
+              create,
+              diclosedContract.metadata.createdAt,
+              diclosedContract.metadata.driverMetadata,
+            )
+          }
```

Comment:
The engine is now responsible to add the meta data to the used disclosed contracts. There is no reason to pass those info to Speedy.